cloudferry_devlab/cloudferry_devlab/tests/testcases/test_cinder_migration.py

```diff
@@ -140,6 +142,65 @@ def check_mount_point_exists(ip, vol):
             for _file in volume['write_to_file']:
                 check_file_valid(volume['mount_point'] + _file['filename'])
 
+    def test_cinder_volumes_deleted_not_attached_to_vm(self):
```

Comment:
This method looks unreasonably complicated and cries "Please refactor!". Do you really need calling `switch_user()` 3 times? There's also some magic code which tries to deal with tenant names being absent or case insensitive, I'd suggest moving that logic to a separate class/method and reusing it everywhere, it would be very easy to forget about those subtle cases.